internal/controller/accessrequest_controller.go

```diff
@@ -156,6 +162,46 @@ func (r *AccessRequestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return result, nil
 }
 
+type AccessRequestConflictError struct {
+	message string
+}
+
+func (e *AccessRequestConflictError) Error() string {
+	return e.message
+}
+
+func NewAccessRequestConflictError(msg string) *AccessRequestConflictError {
+	return &AccessRequestConflictError{
+		message: msg,
+	}
+}
+
+// Validate will verify if there are existing AccessRequests for the same
+// user/app/role already in progress.
+func (r *AccessRequestReconciler) Validate(ctx context.Context, ar *api.AccessRequest) error {
+	arList, err := r.findAccessRequestsByUserApp(ctx,
+		ar.GetNamespace(),
+		ar.Spec.Subject.Username,
+		ar.Spec.Application.Name,
+		ar.Spec.Application.Namespace)
+	if err != nil {
+		return fmt.Errorf("error finding AccessRequests by user and app: %w", err)
+	}
+	for _, arResp := range arList.Items {
+		// skip if it is the same AccessRequest or if the role is different
+		if (arResp.GetName() == ar.GetName() &&
+			arResp.GetNamespace() == ar.GetNamespace()) ||
+			arResp.Spec.RoleTemplateName != ar.Spec.RoleTemplateName {
+			continue
+		}
```

Comment:
```suggestion
		// skip if it is the same AccessRequest
		if arResp.GetName() == ar.GetName() &&
			arResp.GetNamespace() == ar.GetNamespace() {
			continue
		}
		// skip if the requests are for different role templates
		if arResp.Spec.RoleTemplateName != ar.Spec.RoleTemplateName {
			continue
		}
```

This works better for my brain.

Reply:
agreed! I wasn't happy for that if either! tks!

Reply:
done